pkg/controller/clone-controller.go

```diff
@@ -624,6 +624,7 @@ func MakeCloneSourcePodSpec(sourceVolumeMode corev1.PersistentVolumeMode, image,
 					VolumeSource: corev1.VolumeSource{
 						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
 							ClaimName: sourcePvcName,
+							ReadOnly:  sourceVolumeMode == corev1.PersistentVolumeBlock,
```

Comment:
isn't it a little confusing to read? the impression is that we set readOnly: true only for block, but that's not the case
previously we had it as the following:
https://github.com/kubevirt/containerized-data-importer/blob/4d4ad12df5673c847f2bf232607f395faf8b8712/pkg/controller/clone-controller.go#L673
and https://github.com/kubevirt/containerized-data-importer/blob/4d4ad12df5673c847f2bf232607f395faf8b8712/pkg/controller/clone-controller.go#L732-L736

Reply:
So I am reading the `pod.spec.volumes.persistentVolumeClaim` API wrong?
I thought it is just about enforcing the lower level readOnly fields on the container.
```readOnly Will force the ReadOnly setting in VolumeMounts. Default false.```

I mean we should go with https://github.com/kubevirt/containerized-data-importer/blob/4d4ad12df5673c847f2bf232607f395faf8b8712/pkg/controller/clone-controller.go#L670-L673
(As we have previously, no container level setting at all)